tests/components/backup/test_websocket.py

```diff
@@ -2425,13 +2389,18 @@ async def test_config_retention_copies_logic(
         "minor_version": store.STORAGE_VERSION_MINOR,
     }
     get_backups.return_value = (backups, get_backups_agent_errors)
-    delete_backup.return_value = delete_backup_agent_errors
     await hass.config.async_set_time_zone("Europe/Amsterdam")
     freezer.move_to("2024-11-11 12:00:00+01:00")
 
-    await setup_backup_integration(hass, remote_agents=["test-agent"])
+    await setup_backup_integration(hass, remote_agents=["test-agent", "test-agent2"])
     await hass.async_block_till_done()
 
+    manager = hass.data[DATA_MANAGER]
+    for agent_id, agent in manager.backup_agents.items():
+        agent.async_delete_backup = AsyncMock(
+            side_effect=agent_delete_backup_side_effects.get(agent_id), autospec=True
+        )
```

Comment:
In a follow-up, I think we should:
- Make the mock agents mocks so we don't need to assign mocks here
- Make the `setup_backup_integration` return the mock agents so we don't need to access the manager to get references to them